### PR TITLE
Model of a draft tube components (a cone and an elbow)

### PR DESCRIPTION
**Pull request by pandeysudan27**
The previous model of the draft tube was DraftTube.mo. That model is now deleted and new package Drafttube is consider inside water ways containing model of components, a cone and an elbow, of the drafttube for hydro power system.

### CODE REVIEW

**Comment by pandeysudan27** [comment]
I forgot to include the documentation. We should hold this pull request.

**Review by dietmarw — CHANGES_REQUESTED** [review]
Here is a list of changes that need to be done before it can be merged:
 1. ~~The branch needs rebasing on the latest master~~
 2. no more png images. All images shall be SVG
 3. no bitmap icons, please draw the icons using graphical annotations
 4. We can not simply remove the existing `DraftTube` since several user models already use that model. We can however upgrade old models via conversion script from using `DraftTube` to for example `DraftTubeCone` (see below regarding the name). Question is, is the new cone model basically a replacement for the old `DraftTube` model? 
 5. Don't introduce a new package just for those two variants. Simply have them `DraftTubeCone` and `DraftTubeElbow` on the same level of the existing one.

**Comment by dietmarw** [comment]
I've done already the rebase now. Also CI checks are thus activated.

File: OpenHPL/Waterway/Drafttube/Cone.mo
Comment (dietmarw) [inline OpenHPL/Waterway/Drafttube/Cone.mo:80]: @pandeysudan27 When looking at the implementation I was wondering where these coefficients come from and valid they are for the broader application. Can you use them universally?

File: OpenHPL/Waterway/Drafttube/Elbow.mo
Comment (dietmarw) [inline OpenHPL/Waterway/Drafttube/Elbow.mo:83]: @pandeysudan27 The same again here, are these coefficients specific for one geometry or universal?

File: OpenHPL/Waterway/Drafttube/Cone.mo
Comment (pandeysudan27) [reply OpenHPL/Waterway/Drafttube/Cone.mo:80]: Hi Dietmar, these coefficients are calculated based on the non-dimensional plots from the handbook of hydraulic resistance. From my experience, I think we can use them universally if it has asymmetric size as explained in the handbook. I need to test experimental data with this model.

File: OpenHPL/Waterway/Drafttube/Elbow.mo
Comment (pandeysudan27) [reply OpenHPL/Waterway/Drafttube/Elbow.mo:83]: These expressions are also calculated from the hydraulic resistance handbook based on ratios of inner curvature and the bend angle of the elbow.

File: OpenHPL/Waterway/Drafttube/Cone.mo
Comment (dietmarw) [reply OpenHPL/Waterway/Drafttube/Cone.mo:80]: In that case could you document this with a link to the source in the info layer?

Comment (dietmarw) [reply OpenHPL/Waterway/Drafttube/Cone.mo:80]: Actually if these values are calculated why not put the calculation in here with the parameters you used as input parameters.

File: OpenHPL/Waterway/Drafttube/Elbow.mo
Comment (dietmarw) [reply OpenHPL/Waterway/Drafttube/Elbow.mo:83]: Again, maybe better to put the original calculation with the input parameters in here.

**Comment by dietmarw** [comment]
Looking a bit further into this there is **a lot** of duplicated code here. Putting several equations in one line makes the comparison harder than it needs to be. So please never put multiple definitions or equations into one line. 

Now to me it looks like these two models should rather be combined to one with  a parameter switch for the shape since there are just a couple of calculations different. At the same time you should not point to things that need to be added later in the comment. Do them now **before**  they get merged into the master and hence the next version. 

Also it would be important to note what is wrong with the existing DraftTube model. That one already had a switching parameter. So maybe you should just extend that one with the two new shapes?

File: OpenHPL/Waterway/Drafttube/Cone.mo
Comment (pandeysudan27) [reply OpenHPL/Waterway/Drafttube/Cone.mo:80]: I have documented a draft paper for modeling of cone, elbow and a diffuser. But we have to extend this work upto cavitation prediction which I plan to work on later months. May be the model considering center of mass or without considering may not effect the solution of ODEs, and has to be tested on exp data.

**Comment by pandeysudan27** [comment]
Which way of creating models is better: DraftTube(package)>>(Cone.mo, Elbow.mo) or DraftTube.mo(if type==cone or elbow or diffuser) ? I have struggle a lot for the later case. Reasons i) too much unnecessary algebraic variables of another type ii) very hard to debug iii) model extension of one type automatically desire equations extension to another type unlike extension of an object inside the model.
 I found this difficulty during modeling for different types of surge tank. It was really hard to extend the equations of ACST. Even to test small case it was difficult. 
Can we stick with former modeling style at it is more easy to group hydro unit models and more important, the model refinememt is easier. This way a completely new unit types can be researched. 
Can you give it a look to SurgeTank.mo for different types of surge tank? If you see there, for the variables and equations that are not in ACST should must be defined in Simple type. Is there a different  modeling technique in Modelica that I am unaware I want to learn it if you could help me.

**Comment by dietmarw** [comment]
@pandeysudan27 Any more development on the draft tube models?

**Comment by pandeysudan27** [comment]
@dietmarw This model of the draft tube din't worked with the experimental data. I am trying with another concept.

**Comment by dietmarw** [comment]
OK.

**Pull request closed without merging**
